app/src/main.rs

```diff
@@ -1,7 +1,7 @@
 extern crate base;
 extern crate middle;
 
-use base::do_a_bunch_of_things;
+use base::{do_a_bunch_of_things,AsRef,as_ref};
```

Comment:
This change may also be confusing the compiler, because it's not a valid import.

Reply:
oops sorry, well i have also tried wthout this change. Actually I made this change only while debugging the error.